TikTokTrainer/UI/CameraView.swift

```diff
@@ -91,13 +98,36 @@ struct CameraView: View {
 
     var body: some View {
         ZStack {
+            if isCountingDown {
+                ZStack {
+                    Rectangle()
+                        .fill()
+                        .foregroundColor(.black)
+                        .opacity(opacity)
+                        .ignoresSafeArea(.all, edges: .all)
+                        .onAppear {
+                            withAnimation {
+                                self.opacity = 0.8
+                            }
+                        }
             CameraPreview(currentImage: $camera.currentUIImage,
                           result: $camera.currentResult,
                           orientation: $camera.currentOrientation)
                 .ignoresSafeArea(.all, edges: .all)
                 .onTapGesture(count: 2) {
                     camera.switchCameraInput()
                 }.zIndex(-1)
+                }
+            } else {
+                CameraPreview(currentImage: $camera.currentUIImage,
+                              result: $camera.currentResult,
+                              orientation: $camera.currentOrientation)
+                    .ignoresSafeArea(.all, edges: .all)
+                    .onTapGesture(count: 2) {
+                        camera.switchCameraInput()
+                    }.zIndex(-1)
```

Comment:
I'd move this to a member of the struct and call it that way instead of repeating the same code twice. That way if we want to change the preview we can just do it in one place.

Reply:
Yeah, good call. I'll update it tomorrow